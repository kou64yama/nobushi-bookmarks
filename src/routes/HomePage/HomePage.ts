import Vue from 'vue';
import Component from 'vue-class-component';

export class HomePage extends Vue {
  public mounted() {
    this.$store.dispatch('document/setTitle', 'Home');
  }
}

export default Component({})(HomePage);
