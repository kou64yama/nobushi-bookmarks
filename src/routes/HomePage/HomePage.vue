<template>
  <v-container fluid class="pa-0">
    <item-list/>
  </v-container>
</template>

<script lang="ts" src="./HomePage.ts"></script>
