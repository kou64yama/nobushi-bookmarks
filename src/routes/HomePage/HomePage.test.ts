import Vuex, { Store, Dispatch } from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import HomePage from './HomePage.vue';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('HomePage', () => {
  const createStore = () =>
    Object.assign(new Vuex.Store({}), {
      dispatch: jest.fn(),
    });

  it('should render correctly', () => {
    const store = createStore();
    const wrapper = shallowMount(HomePage, {
      localVue,
      store,
      stubs: {
        VContainer: true,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(store.dispatch.mock.calls[0][0]).toEqual('document/setTitle');
  });
});
