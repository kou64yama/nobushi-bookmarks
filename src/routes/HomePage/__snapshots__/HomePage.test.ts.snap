// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render correctly 1`] = `
<vcontainer-stub
  class="pa-0"
  fluid=""
>
  <itemlist-stub />
</vcontainer-stub>
`;
